### DISCUSSION TITLE
How to use the http/sock proxy

### DISCUSSION THREAD

**Discussion by lyyyuna** [XAMPPRocky/octocrab#368] [Q&A] [answered]
My client is behind a restricted network, it requires a proxy to connect to the GitHub.

**Comment by XAMPPRocky** [comment, marked as answer]
Thank you for your question, you can change the `base_uri` in the settings. https://docs.rs/octocrab/latest/octocrab/struct.OctocrabBuilder.html#method.base_uri

```rust
let octocrab = octocrab::OctocrabBuilder::default()
    .add_preview("machine-man")
    .base_uri("https://github.example.com")?
    .build()?;
```

**Reply by lyyyuna** [reply]
Still confused, where to put the http proxy addr in the `base_uri`?

**Reply by XAMPPRocky** [reply]
You set your proxy address as the `base_uri`? I would need more information about the proxy setup in order to help you further.

**Reply by lyyyuna** [reply]
For example, i have a http proxy `127.0.0.1:8080`,  which will direct all github api request to `https://api.github.com`.

From the doc, the `base_uri` seems to change the github api address, not to set the proxy?

**Reply by XAMPPRocky** [reply]
> seems to change the github api address, not to set the proxy?

As far as I can tell from what you're saying those are the same thing, since any request to `127.0.0.1:8080` will end up at `https://api.github.com` so `127.0.0.1:8080` is the API address.
